x/csr/keeper/evm_hooks.go

```diff
@@ -109,7 +111,18 @@ func (h Hooks) processEvents(ctx sdk.Context, receipt *ethtypes.Receipt) error {
 				return err
 			}
 		case nftAddress:
-			continue
+			// retrieve the event emitted from CSR NFT
+			event, err := csrNftContract.EventByID(eventID)
+			if err != nil {
+				return err
+			}
+
+			// only the Withdrawal Event can be emitted from CSRNFT,
+			switch event.Name {
+			case types.WithdrawalEvent:
+				// handle withdrawal
+				err = h.k.WithdrawalEvent(ctx, log.Data)
+			}
```

Comment:
```suggestion
			}
			if err != nil {
			 // return error
			}
```
You are not returning the error here